src/Npgsql/Internal/Resolvers/ExtraConversionsResolver.cs

```diff
@@ -0,0 +1,130 @@
+using System;
+using Npgsql.Internal.Converters;
+using Npgsql.PostgresTypes;
+
+namespace Npgsql.Internal.Resolvers;
+
+sealed class ExtraConversionsResolver : IPgTypeInfoResolver
```

Comment:
All of these probably need to be unsealed to support the inheritance pattern to opt into arrays (though I can't say for sure, it'll end up working like that)

Reply:
Ah, ok. Up to you - if you want I can remove them now, or you can do it when you finalize the array stuff.